Zend/zend_compile.c

```diff
@@ -5506,17 +5504,38 @@ void zend_compile_closure_uses(zend_ast *ast) /* {{{ */
 }
 /* }}} */
 
+void zend_check_attr(uint32_t attr, const char* entity){ /* {{{ */
+	if (memcmp(entity, "__callStatic", 12) == 0){
+		if(!(attr & ZEND_ACC_PUBLIC) || !(attr & ZEND_ACC_STATIC)){
+			zend_error(E_WARNING, "The magic method __callStatic() must have public visibility and be static");
+		}
+	} else if (entity[0] == '_' && entity[1] == '_'
+				&& (!(attr & ZEND_ACC_PUBLIC) || (attr & ZEND_ACC_STATIC))){
+		zend_error(E_WARNING,
+				"The magic method %s() must have public visibility and cannot be static",
+				entity);
+	} else if(memcmp(entity, "constant", 8) == 0|| memcmp(entity, "method", 6) == 0){
```

Comment:
I don't think this should be combined with the magic method checking checking code.